### PR TITLE
Replaced dealii::Vector with ArrayView in internal FEValues calls.

### PR DESCRIPTION
**Pull request by luca-heltai**
Relates to #4706 and #4704

### CODE REVIEW

File: source/fe/fe_values.cc [hunk 1]
```diff
@@ -1304,7 +1306,7 @@ namespace FEValuesViews
     dealii::Vector<typename InputVector::value_type> dof_values (fe_values->dofs_per_cell);
     fe_values->present_cell->get_interpolated_dof_values(fe_function, dof_values);
     internal::do_function_derivatives<1,dim,spacedim>
-    (dof_values, fe_values->finite_element_output.shape_gradients, shape_function_data, gradients);
+    (ArrayView<const typename InputVector::value_type>(&dof_values[0], fe_values->dofs_per_cell), fe_values->finite_element_output.shape_gradients, shape_function_data, gradients);
```
Comment (bangerth) [inline source/fe/fe_values.cc]: break this line and similar ones

File: source/fe/fe_values.cc [hunk 2]
```diff
@@ -471,7 +472,7 @@ namespace FEValuesViews
            shape_function<dofs_per_cell; ++shape_function)
         if (shape_function_data[shape_function].is_nonzero_shape_function_component)
           {
-            const Number value = dof_values(shape_function);
+            const Number value = dof_values[shape_function];
```
Comment (jppelteret) [inline source/fe/fe_values.cc]: Preferentially use an alias here instead of creating a temporary. Same below

Comment (bangerth) [inline source/fe/fe_values.cc]: For scalars, temps are actually simpler for the compiler. Are you concerned about things such as AD types?

**Comment by drwells** [comment]
We could make this a little nicer by writing

```cpp
make_array_view(dof_values.begin(), dof_values.end())
```

instead of typing out the full `ArrayView<typename InputVector::value_type>(...)`.

**Comment by kronbichler** [comment]
Looks nice!

**Pull request merged**